src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporter.cs

```diff
@@ -168,15 +170,15 @@ private TestProgressState GetOrAddAssemblyRun(string assembly, string? targetFra
         return _assemblies.GetOrAdd(key, _ =>
         {
             IStopwatch sw = CreateStopwatch();
-            var assemblyRun = new TestProgressState(assembly, targetFramework, architecture, sw);
+            var assemblyRun = new TestProgressState(_counter++, assembly, targetFramework, architecture, sw);
```

Comment:
For all the places where you do `_counter++`, shall we interlocked them instead to ensure no duplicates?

Reply:
@nohwnd is the TerminalTestReporter called in parallel?

Reply:
not here but it is in dotnet test. So interlocked would be better.